deeprank2/dataset.py

```diff
@@ -429,28 +434,28 @@ def __init__( # pylint: disable=too-many-arguments
             hdf5_path (Union[str,list]): Path to .HDF5 file(s). For multiple .HDF5 files, insert the paths in a List. Defaults to None.
             subset (Optional[List[str]], optional): List of keys from .HDF5 file to include. Defaults to None (meaning include all).
             train (bool, optional): Boolean flag to determine if the instance represents the training set.
-                If False, a dataset_train of the same class must be provided as well.
-                The latter will be used to scale the validation/testing set according to its features values and to match the datasets' parameters.
+                It needs to be set to False only if the instance represents the validation or testing set.
                 Defaults to True.
-            dataset_train (class:`GridDataset`, optional): If `train` is True, assign here the training set.
-                If `train` is False and `dataset_train` is assigned,
-                the parameters `features`, `target`, `traget_transform`, `task`, and `classes` will be inherited from `dataset_train`.
+            train_data (Optional[Union[str, class:`GridDataset`]], optional)): data to inherit information from the training dataset or the pre-trained model.
+                It is considered only if `train` is False. In the latter case, `train_data` needs to be a dataset of the same class or
+                the path of a DeepRank2 pre-trained model. If set, the parameters `features`, `target`, `traget_transform`, `task`, and `classes`
+                will be inherited from `train_data`.
```

Comment:
I like the idea of using only one parameter (maybe we can think of a better naming?) that can be either None, a pre-trained model, or a dataset_train. It seems much easier to me also code-wise, and less redundant. Let me know what you think about this option, and if you have an idea for better naming (maybe `train_source`? Or maybe `train_data` is good enough)

Reply:
@DaniBodor

Reply:
We agreed to remove the current `train` argument, as it is redundant with the current `train_data` argument. If `train_data is None`, that means that means the same as `train is True`. Otherwise it means `train is False`.

Also, we agreed to rename it to `train_source` until we come up with a better name for the parameter.